govtech/detectors/cves/cve_2020_3452/src/test/java/com/google/tsunami/plugins/detectors/cves/cve_2020_3452/CVE20203452DetectorTest.java

```diff
@@ -0,0 +1,198 @@
+    ImmutableList<NetworkService> httpServices =
+        ImmutableList.of(
+            NetworkService.newBuilder()
+                .setNetworkEndpoint(
+                    forHostnameAndPort(mockWebServer.getHostName(), mockWebServer.getPort()))
+                .setTransportProtocol(TransportProtocol.TCP)
+                .setSoftware(Software.newBuilder().setName("Cisco Adaptive Security Appliance"))
+                .setServiceName("http")
+                .build());
+
+    assertThat(
+            detector
+                .detect(buildTargetInfo(forHostname(mockWebServer.getHostName())), httpServices)
+                .getDetectionReportsList())
+        .isEmpty();
+  }
+
+  @Test
+  public void detect_whenNonCiscoASAWebApp_ignoresServices() throws IOException {
+    startMockWebServer(
+        "+CSCOT+/translation-table?type=mst&textdomain=/%2bCSCOE%2b/../../../../..//"
+        + "%2bCSCOE%2b/include/browser_inc.lua&default-language&lang=../",
+        HttpStatus.OK.code(),
+        "This is WordPress.");
+    ImmutableList<NetworkService> httpServices =
+        ImmutableList.of(
+            NetworkService.newBuilder()
+                .setNetworkEndpoint(
+                    forHostnameAndPort(mockWebServer.getHostName(), mockWebServer.getPort()))
+                .setTransportProtocol(TransportProtocol.TCP)
+                .setSoftware(Software.newBuilder().setName("WordPress"))
+                .setServiceName("http")
+                .build());
+
+    assertThat(
+            detector
+                .detect(buildTargetInfo(forHostname(mockWebServer.getHostName())), httpServices)
+                .getDetectionReportsList())
+        .isEmpty();
+  }
+
+  @Test
+  public void detect_whenNonHttpNetworkService_ignoresServices() {
+    ImmutableList<NetworkService> nonHttpServices =
+        ImmutableList.of(
+            NetworkService.newBuilder().setServiceName("ssh").build(),
+            NetworkService.newBuilder().setServiceName("rdp").build());
+    assertThat(
+            detector
+                .detect(buildTargetInfo(forHostname(mockWebServer.getHostName())), nonHttpServices)
+                .getDetectionReportsList())
+        .isEmpty();
+  }
+
+  @Test
+  public void detect_whenEmptyNetworkService_generatesEmptyDetectionReports() {
+    assertThat(
+            detector
+                .detect(
+                    buildTargetInfo(forHostname(mockWebServer.getHostName())), ImmutableList.of())
+                .getDetectionReportsList())
+        .isEmpty();
+  }
+
+  private void startMockWebServer(String url, int responseCode, String response)
+      throws IOException {
+    mockWebServer.enqueue(new MockResponse().setResponseCode(responseCode).setBody(response));
+    mockWebServer.start();
+    mockWebServer.url(url);
+  }
+
+  private static TargetInfo buildTargetInfo(NetworkEndpoint networkEndpoint) {
+    return TargetInfo.newBuilder().addNetworkEndpoints(networkEndpoint).build();
+  }
+}
```

Comment:
Please add a new line at the end of this file.

Reply:
Done!

Reply:
Done!